micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemoryMetrics.java

```diff
@@ -100,6 +98,58 @@ public void bindTo(MeterRegistry registry) {
                 .baseUnit(BaseUnits.BYTES)
                 .register(registry);
         }
+
+        //The used and committed size of the returned memory usage is the sum of those values of all heap/non-heap memory pools
+        // whereas the max size of the returned memory usage represents the setting of the non-heap/non-heap memory which may
+        // not be the sum of those of all non-heap memory pools. If the setting is missing for heap/non-heap memory pools these values come as -1.
+
+        MemoryMXBean memoryMXBean = ManagementFactory.getMemoryMXBean();
+
+        Iterable<Tag> tagsWithId = Tags.concat(tags,"id", "total", "area", "heap");
+
+        Gauge.builder("jvm.memory.used", memoryMXBean, (mem) -> getTotalAreaUsageValue(mem, MemoryUsage::getUsed, "heap"))
```

Comment:
why sum these up? dimensional monitoring systems will allow you to select all individual heap areas and sum over them as part of the query?

Reply:
We are using Statsd meter registry to send metrics to InfluxDB and we use Grafana for visualising these metrics. As you can see in the screenshot attached, we are seeing some missing points in the Grafana panels when we are running queries to sum over the individual heap areas. 

![Grafana_JVM_Heap_Committed_Used](https://user-images.githubusercontent.com/56631243/88126005-329c9900-cbee-11ea-828a-f153ee9a8cc4.png)

Same issue is seen in our alerting mechanism, Kapacitor also. We are seeing wrong alerts due to any missing points of individual heap areas. To avoid these issues, it would be better to take these values directly from MemoryMXBean without any calculations.

Reply:
Can you shed some light on why there are missing points? Telegraf dropping them? What's going on with this?

Reply:
We are investigating on whether missing points are at the Telegraf side or at statsd push. Allow me sometime to get back on this.